### ISSUE TITLE
defaultValue: true

### ISSUE THREAD

**Issue by berlin75** [sequelize/sequelize#11121]
## What are you doing?

```js
var Foo = sequelize.define('foo', {
  name: Sequelize.STRING,
  flag: { type: Sequelize.BOOLEAN, allowNull: false, defaultValue: true}
})
Foo.sync().then(() => {
  Foo.create({name: "name"})
})
```

## What do you expect to happen?
set the default value to true

## What is actually happening?
nodejs output: 
Unhandled rejection SequelizeDatabaseError: Incorrect integer value: 'true' for column 'flag' at row 1

mysql log:
INSERT INTO `foos` (`id`,`name`,`flag`,`createdAt`,`updatedAt`) VALUES (DEFAULT,'name','true','2019-06-27 22:41:25','2019-06-27 22:41:25')

boolean true was converted to string 'true' automatically by sequelize

## Environment
Dialect: mysql
Database version: 5.7.9
Sequelize version: 5.8.11
Node Version: v8.9.1
OS: win10

**Comment by pucumt** [comment]
the same issue in Sequelize 5.10.0

**Comment by pucumt** [comment]
The issue will be resolved once mysql2 upgrade to 1.6.5

**Comment by papb** [comment]
> The issue will be resolved once mysql2 upgrade to 1.6.5

Can you clarify this? I did not understand. Apparently mysql2 version 1.6.5 has been around for quite a while...

**Comment by sushantdhiman** [comment]
Looks invalid see https://github.com/sequelize/sequelize/pull/12090#issuecomment-611631523
